File: Houseclub/src/main/java/me/grishka/houseclub/api/methods/UpdateUsername.java
package me.grishka.houseclub.api.methods;

import me.grishka.houseclub.api.BaseResponse;
import me.grishka.houseclub.api.ClubhouseAPIRequest;

public class UpdateUsername extends ClubhouseAPIRequest<BaseResponse> {
    public UpdateUsername(String name) {
        super("POST", "update_username", BaseResponse.class);
        requestBody = new Body(name);
    }

    private static class Body {
        public String username;

        public Body(String username) {
            this.username = username;
        }
    }
}

File: Houseclub/src/main/java/me/grishka/houseclub/api/methods/UpdateUsername.kt
package me.grishka.houseclub.api.methods

import me.grishka.houseclub.api.BaseResponse
import me.grishka.houseclub.api.ClubhouseAPIRequest

class UpdateUsername(name: String) :
    ClubhouseAPIRequest<BaseResponse?>("POST", "update_username", BaseResponse::class.java) {
    private class Body(var username: String)

    init {
        requestBody = Body(name)
    }
}